services/subnet_manager/src/main/java/com/futurewei/alcor/subnet/service/implement/SubnetServiceImp.java

```diff
@@ -319,6 +319,26 @@ public boolean verifyCidrBlock(String cidr) throws FallbackException {
 
     }
 
+    @Override
+    @DurationStatistics
+    public Integer getUsedIpByRangeId(String rangeId) throws UsedIpsIsNotCorrect {
+        String ipManagerServiceUrl = ipUrl + "range" + "/" + rangeId;
+        IpAddrRangeRequest ipAddrRange = restTemplate.getForObject(ipManagerServiceUrl, IpAddrRangeRequest.class);
+        if (ipAddrRange == null) {
+            logger.info("can not find ipAddrRange by range id");
+            return null;
+        }
+
+        Long usedIPs = ipAddrRange.getUsedIps();
+        if (usedIPs == null || usedIPs > Integer.MAX_VALUE || usedIPs < 0) {
+            throw new UsedIpsIsNotCorrect();
+        }
+
+        Integer usedIps = Integer.parseInt(String.valueOf(usedIPs));
```

Comment:
if userdIPs is Long, why do we want to convert it to String, then to Integer?

Reply:
In openstack, usedIps is integer, but in private ip manager, usedIps is Long

Reply:
Okay, we can directly convert Long to Integer without an intermediate step of converting it to String, right? Also, we will need to handle overflow situation.